crowbar_framework/lib/repository.rb

```diff
@@ -0,0 +1,144 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Repository
+  class Check
```

Comment:
I still think this is the wrong OO design.  The object should be a repository, and the check should be an action (i.e. method).  Objects have state; actions don't.  Also doesn't this class belong in the `Crowbar` namespace?

So:

```ruby
module Crowbar
  class Repository
    ...
    def check
      ...
```